### PR TITLE
(Feature) Added Yii2 Object Storage Component

### PR DESCRIPTION
**Pull request by yemexx1**
Added Yii2 Object Storage component and implemented frosteallth s3 client

### CODE REVIEW

File: src/File/FrostealthS3ObjectStorageClient.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * @author Adeyemi Olaoye <yemi@cottacush.com>
+ * @package CottaCush\Yii2\File
+ */
+class FrostealthS3ObjectStorageClient implements ObjectStorageClientInterface
+{
+    public function put($resource, $fileName)
+    {
+        try {
+            /** @var Storage $s3Bucket */
+            $s3Bucket = Yii::$app->get('s3bucket');
```
Comment (yemexx1) [inline src/File/FrostealthS3ObjectStorageClient.php:22]: @goke-epapa Any ideas on how to parameterize this?

Comment (goke-epapa) [reply src/File/FrostealthS3ObjectStorageClient.php:22]: Not at the moment, let's leave as is for now, we can always revisit. Thanks

File: src/File/FrostealthS3ObjectStorageClient.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            $result = $s3Bucket->put($fileName, $resource);
+            $mediaUrl = ArrayHelper::getValue($result->toArray(), 'ObjectURL');
+            return $mediaUrl;
+        } catch (S3Exception $ex) {
+            Yii::$app->log->logger->log($ex->getMessage(), Logger::LEVEL_ERROR);
```
Comment (goke-epapa) [inline src/File/FrostealthS3ObjectStorageClient.php:27]: I think you should add `$category`  for the log so we can track it in papertrail. It should be the third parameter for the method.

**Review by goke-epapa — CHANGES_REQUESTED** [review]
Just a slight change

**Pull request closed without merging**